### PR TITLE
Length is a monoid homomorphism

### LINKED ISSUE #1857: `List.length` is a monoid homomorphism
We should probably also add raw magma and monoid bundles to `Data.List.Base` as an extension to #1755.

### PR DESCRIPTION
**Pull request by Taneb**
Also adds raw bundles in Data.List.Base. Closes #1857

### CODE REVIEW

**Comment by jamesmckinna** [comment]
Should that new definition in `Data.List.Base` not be `++-[]-rawMonoid`?

**Comment by Taneb** [comment]
You're exactly right. Serves me right for copy-pasting to save typing :sweat_smile:

**Comment by jamesmckinna** [comment]
Similarly for `length-isMonoidHomomorphism : IsMonoidHomomorphism (++-[]-rawMonoid A) +-0-rawMonoid length`?

File: src/Data/List/Base.agda [hunk 1]
```diff
@@ -463,6 +465,24 @@ infixl 6 _∷ʳ?_
 _∷ʳ?_ : List A → Maybe A → List A
 xs ∷ʳ? x = maybe′ (xs ∷ʳ_) xs x
 
+------------------------------------------------------------------------
+-- Raw algebraic bundles
+
+module _ (A : Set a) where
```
Comment (MatthewDaggitt) [inline src/Data/List/Base.agda:471]: Very minor but can we have an empty line between the module declarations and their contents? I've added it to the list of things to add to the style guide in #1018

**Pull request merged**